# Milestone-1 Assignment - Anita Aifesehi

### Data Analysis

In [13]:
# Import pandas 

import pandas as pd 

In [14]:
df = pd.read_csv(r"C:\Users\anita\OneDrive\Documents\BOOTCAMP_DATA_ANALYTICS\ocd_patient_dataset.csv")

In [15]:
# To check for top 5 of dataset
df.head()

,Patient ID,Age,Gender,Ethnicity,Marital Status,Education Level,OCD Diagnosis Date,Duration of Symptoms (months),Previous Diagnoses,Family History of OCD,Obsession Type,Compulsion Type,Y-BOCS Score (Obsessions),Y-BOCS Score (Compulsions),Depression Diagnosis,Anxiety Diagnosis,Medications
0,1018,32,Female,African,Single,Some College,2016-07-15,203,MDD,No,Harm-related,Checking,17,10,Yes,Yes,SNRI
1,2406,69,Male,African,Divorced,Some College,2017-04-28,180,NaN,Yes,Harm-related,Washing,21,25,Yes,Yes,SSRI
2,1188,57,Male,Hispanic,Divorced,College Degree,2018-02-02,173,MDD,No,Contamination,Checking,3,4,No,No,Benzodiazepine
3,6200,27,Female,Hispanic,Married,College Degree,2014-08-25,126,PTSD,Yes,Symmetry,Washing,14,28,Yes,Yes,SSRI
4,5824,56,Female,Hispanic,Married,High School,2022-02-20,168,PTSD,Yes,Hoarding,Ordering,39,18,No,No,NaN


In [5]:
# To check for missing values
df.isnull().sum()

Patient ID                         0
Age                                0
Gender                             0
Ethnicity                          0
Marital Status                     0
Education Level                    0
OCD Diagnosis Date                 0
Duration of Symptoms (months)      0
Previous Diagnoses               248
Family History of OCD              0
Obsession Type                     0
Compulsion Type                    0
Y-BOCS Score (Obsessions)          0
Y-BOCS Score (Compulsions)         0
Depression Diagnosis               0
Anxiety Diagnosis                  0
Medications                      386
dtype: int64

In [26]:
# Filling in the missing values with 'Unspecified'
df['Previous Diagnoses'] = df['Previous Diagnoses'].fillna('Unspecified')

In [28]:
df['Medications'] = df['Medications'].fillna('Unspecified')

In [30]:
# To check for duplicates
print("Duplicates:", df.duplicated().sum())

Duplicates: 0


In [9]:
# To check for columns and rows
print(df.shape)

(1500, 17)


In [10]:
# To check the data types
df.dtypes

Patient ID                        int64
Age                               int64
Gender                           object
Ethnicity                        object
Marital Status                   object
Education Level                  object
OCD Diagnosis Date               object
Duration of Symptoms (months)     int64
Previous Diagnoses               object
Family History of OCD            object
Obsession Type                   object
Compulsion Type                  object
Y-BOCS Score (Obsessions)         int64
Y-BOCS Score (Compulsions)        int64
Depression Diagnosis             object
Anxiety Diagnosis                object
Medications                      object
dtype: object

In [11]:
# To check the data information 
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1500 entries, 0 to 1499
Data columns (total 17 columns):
 #   Column                         Non-Null Count  Dtype 
---  ------                         --------------  ----- 
 0   Patient ID                     1500 non-null   int64 
 1   Age                            1500 non-null   int64 
 2   Gender                         1500 non-null   object
 3   Ethnicity                      1500 non-null   object
 4   Marital Status                 1500 non-null   object
 5   Education Level                1500 non-null   object
 6   OCD Diagnosis Date             1500 non-null   object
 7   Duration of Symptoms (months)  1500 non-null   int64 
 8   Previous Diagnoses             1500 non-null   object
 9   Family History of OCD          1500 non-null   object
 10  Obsession Type                 1500 non-null   object
 11  Compulsion Type                1500 non-null   object
 12  Y-BOCS Score (Obsessions)      1500 non-null   int64 
 13  Y-B

In [35]:
#  df.describe() with written commentary in plain English 

df.describe()

,Patient ID,Age,Duration of Symptoms (months),Y-BOCS Score (Obsessions),Y-BOCS Score (Compulsions)
count,1500.000000,1500.000000,1500.000000,1500.000000,1500.00000
mean,5541.254000,46.781333,121.745333,20.048000,19.62600
std,2562.389469,16.830321,67.404610,11.823884,11.78287
min,1017.000000,18.000000,6.000000,0.000000,0.00000
25%,3338.000000,32.000000,64.000000,10.000000,9.00000
50%,5539.500000,47.000000,121.000000,20.000000,20.00000
75%,7745.500000,61.000000,178.000000,31.000000,29.00000
max,9995.000000,75.000000,240.000000,40.000000,40.00000


### Observations
The dataset contains 1,500 observations, and the numeric columns shown all have a count of 1,500, suggesting there are no missing values in these columns. The average patient age is approximately 46.8 years, while the median is 47 years, with ages ranging from 18 to 75. The average Y-BOCS scores for obsessions and compulsions are about 20, and both range from 0 to 40, indicating a wide range of OCD symptom severity in the dataset.

In [43]:
# At least 3 GroupBy analyses with a one-sentence business insight for each 

age_gender = df.groupby('Gender')['Age'].mean()
print(age_gender)

Gender
Female    46.933066
Male      46.630810
Name: Age, dtype: float64


Insight: The average age of female patients is about 46.93 years compared with 46.63 years for male patients, showing almost no age difference between the two gender groups.

In [52]:
ethnicity_score = df.groupby('Ethnicity')['Y-BOCS Score (Obsessions)'].mean()
print(ethnicity_score)

Ethnicity
African      19.759259
Asian        20.316062
Caucasian    19.791457
Hispanic     20.283163
Name: Y-BOCS Score (Obsessions), dtype: float64


Insight: This analysis compares the average severity of obsessive symptoms across different ethnic groups and identifies which group has the highest average obsession score, which is Asian. 

In [56]:
obsession_duration = df.groupby('Obsession Type')['Duration of Symptoms (months)'].mean()
print(obsession_duration)

Obsession Type
Contamination    127.738562
Harm-related     115.372372
Hoarding         124.309353
Religious        120.016502
Symmetry         122.100000
Name: Duration of Symptoms (months), dtype: float64


Insight: This analysis shows which types of obsessions are associated with a longer average duration of symptoms.

In [61]:
# A correlation matrix with commentary on what you found

df.corr(numeric_only=True).round(2)

,Patient ID,Age,Duration of Symptoms (months),Y-BOCS Score (Obsessions),Y-BOCS Score (Compulsions)
Patient ID,1.00,0.03,-0.03,0.02,0.02
Age,0.03,1.00,-0.00,-0.00,0.01
Duration of Symptoms (months),-0.03,-0.00,1.00,0.01,0.04
Y-BOCS Score (Obsessions),0.02,-0.00,0.01,1.00,0.03
Y-BOCS Score (Compulsions),0.02,0.01,0.04,0.03,1.00


The correlation matrix shows very weak or no linear relationships among the numerical variables. The strongest correlation is between duration of symptoms and Y-BOCS compulsion score at 0.04, which is still negligible. Age also shows almost no relationship with symptom duration or OCD severity scores. Overall, none of the numeric variables demonstrate a meaningful linear correlation in this dataset.

In [68]:
# Outlier detection using the IQR method on Age

Q1 = df['Age'].quantile(0.25)
Q3 = df['Age'].quantile(0.75)

IQR = Q3 - Q1

lower_bound = Q1 - 1.5 * IQR
upper_bound = Q3 + 1.5 * IQR

outliers = df[(df['Age'] < lower_bound) | (df['Age'] > upper_bound)]

print(f'Lower bound: {lower_bound:.2f}')
print(f'Upper bound: {upper_bound:.2f}')
print(f'Number of outliers: {len(outliers)}')

print(outliers[['Age']])

Lower bound: -11.50
Upper bound: 104.50
Number of outliers: 0
Empty DataFrame
Columns: [Age]
Index: []


### Analysis Summary

* Dataset Used
For this milestone, I worked with an **OCD patient dataset** containing demographic and clinical information such as age, gender, ethnicity, education level, duration of symptoms, obsession and compulsion types, previous diagnoses, medications, and Y-BOCS obsession and compulsion scores.

* Data Cleaning
I first checked the dataset for missing values, duplicates, incorrect data types, and possible outliers. Missing values were mainly found in the **Previous Diagnoses** and **Medications** columns, so I handled them carefully to avoid unnecessarily removing a large number of patient records. I also reviewed the numerical columns for outliers using the IQR method and checked the consistency of the data before carrying out the analysis.

#### Key Findings
1. The average age of female patients was about **46.93 years**, while the average age of male patients was about **46.63 years**, showing very little difference in age between the two groups.

2. The correlation analysis showed **little to no strong linear relationship** between age, duration of symptoms, and Y-BOCS obsession and compulsion scores. The correlations were all close to zero.

3. Using the IQR method on the **Age** column, I found **no outliers**, meaning the age values were within the expected statistical range and did not require removal.

#### Recommendation
Based on the analysis thus far, factors such as **age and duration of symptoms alone may not be enough to explain OCD symptom severity**. I would recommend looking more closely at other variables such as obsession type, compulsion type, previous diagnoses, medications, and demographic factors to better understand patterns in OCD severity and patient outcomes.